### ISSUE TITLE
SiriusXM Login won't work with email address

### ISSUE THREAD

**Issue by Sparkinman** [music-assistant/hass-music-assistant#3324]
### What version of Music Assistant has the issue?

2.3.3

### What version of the Home Assistant Integration have you got installed?

2024.12.0

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Refuses to login with email address (maybe this works with username, but my credentials are email address only). Can login with the exact same credentials on the SiriusXM website.

### How to reproduce

Add Sirius XM as a music service, enter credentials, and it will timeout.

### Music Providers

SiriusXM

### Player Providers

Sonos

### What version of Home Assistant Core are your running

2024.12.4

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by OzGav** [comment]
So how do you know this is a problem with using the email address?

**Comment by Sparkinman** [comment]
I'm ASSUMING that's the issue since there aren't any other bug reports listed for SiriusXM.
They (SiriusXM) tried maybe 6 months ago to switch me to a username, but at the time sonos required an email instead of a username, so I had them change it back.
Login works fine in the sonos app, works fine on the web, works fine in the app on IOS and Android, just not Music Assistant.

**Comment by OzGav** [comment]
You have not attached the full log as required by the issue template

**Comment by Sparkinman** [comment]
Logs attached, Sorry, I don't normally submit bugs, so forgive me

[d5369777_music_assistant_2024-12-19T23-47-06.097Z.log](https://github.com/user-attachments/files/18204194/d5369777_music_assistant_2024-12-19T23-47-06.097Z.log)

**Comment by OzGav** [comment]
@btoconnor

**Comment by btoconnor** [comment]
I exclusively log in with my email address.  I'm not sure that's the issue.

**Comment by btoconnor** [comment]
To add - it's entirely possible that SiriusXM is doing something funky under the hood with masquerading email addresses as usernames.  As pointed out in other discussions, the `sxm-client` library is using a very old API, and it might be the case that the old API isn't capable of distinguishing between the email and a username.

Is there anyway you can try other values that might be associated as the email address / username on the account?

Maybe check your "Account Info" tab under your Account Information?

Something under here:
![image](https://github.com/user-attachments/assets/1c789769-5647-4dbe-a548-3fdf46501afa)

**Comment by Sparkinman** [comment]
Well, after an hour with siriusXM on the phone, changing the username to a non email address ended up fixing it. I guess we can close this now!
Thanks everyone!

**Comment by btoconnor** [comment]
Glad you were able to sort it out.

I had issues when I first signed up for SiriusXM when they fabricated an email address out of thin air for me - after trying to change it manually, it basically locked me out of the account from logging in, and I had to spend an hour on the phone with them to fix it.  Sounds similar to your issue.

My bet is that they have some really convoluted logic trying to discern usernames and emails, and usernames can sometimes look like emails and the whole thing breaks.  Not so confidence inspiring.
